### ISSUE TITLE
Lint/Void cop false positive

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#6963]
It is the case that `#each` ignores the result of the block for `Array`, `Range` and `Enumerable` and thus creates a void context, `Enumerator#each` does not ignore the result of the block.

```
enum = [1,2,3].map
p enum.each { |_item| 42 } # => [42, 42, 42]
```

I believe that [this cop](https://github.com/rubocop-hq/rubocop/blob/v0.67.2/spec/rubocop/cop/lint/void_spec.rb#L183-L190) is thus overly strict, at least unless the receiver is an array/range literal or is named `array`, `ary`, `range`, etc.

**Comment by marcandre** [comment]
Still valid

**Comment by redencion** [comment]
Hi @marcandre,  What does it should return ?

**Comment by KessaPassa** [comment]
@marcandre 
I am trying of resolving this issue.
I have added debug print `pp node.receiver` 
here: 
https://github.com/rubocop/rubocop/blob/00e7ac78fb83383ef537e9e08ecf2301a1427578/lib/rubocop/cop/lint/void.rb#L66

I tried to run rubocop this code.
```
enum = [1,2,3].map
p enum.each { |_item| 42 } # => [42, 42, 42]
```

Then I have got ` s (: lvar,: enum) `

So, I'd like to check like `return unless node.receive.is_a? (Enumerator)`.

But, I could not know the type of the value of the `node.receiver` .
Because, `node.receiver` have only AST information.
Do you know how to get type of the `node.receiver` ?
